### DISCUSSION TITLE
Discrepancies - Plausible vs Self hosted vs Google Analytics

### DISCUSSION THREAD

**Discussion by MagnusThome** [plausible/analytics#1722] [Self-Hosted Support]
Hi all!

I'm a bit surprised that my self hosted Plausible sees so much less pageviews than the cloud hosted version, I was expecting the opposite? My Google Analytics stats are pretty close to the Plausible hosted stats. My self hosted is lower than both. I have my self hosted plausible set up on the exact same url as the main site but with port 8000 routed to the plausible server. Both are virtual servers and the plausible server seems to have a relaxed life with 4GB ram, NVMe disk and 4 cores from a i7-8700?

![image](https://user-images.githubusercontent.com/32169384/155860613-6d7ec110-11f4-4ec6-b6ea-9938a23dc481.png)

Is this what you normally would see? Or is there something that should/could be done?

![Untitled-1](https://user-images.githubusercontent.com/32169384/155860449-0ccdf162-9461-477e-8751-08297643263f.gif)

**Comment by ukutaht** [comment]
Yes I would also expect more traffic on the self-hosted. We run a self-hosted instance alongside our main environment and we get more traffic on the self-hosted setup

**Reply by ebrtyson** [reply]
Why would the self-hosted setup have more traffic? I would have thought they should be the same...

**Reply by ukutaht** [reply]
because our main domain https://plausible.io is blocked by uBlock and Brave. Probably some other lists as well. Self-hosted should not be blocked as much as long as you don't use obvious tracking domain names like `analytics.mydomain.com`

**Comment by MagnusThome** [comment]
Thanks for the feedback. I just don't get what could be wrong, odd.

**Comment by MagnusThome** [comment]
Could using the outbound script in any way affect the results? I'm also wondering if some browsers are picky about the port 8000. So I changed the url from:

<script defer data-domain="rejsa.nu" src="https://mysiteurl.com:8000/js/plausible.outbound-links.js"></script>
to
<script defer data-domain="rejsa.nu" src="https://plaus.mysiteurl.com/js/plausible.js"></script>  
  
I'll see what it looks like tomorrow :-)

**Comment by MagnusThome** [comment]
Looks like removing the outboundlink and the port 8000 did the trick. Now I just need to test which of them it was :-D

![image](https://user-images.githubusercontent.com/32169384/156669183-4f9ff923-2902-4776-ab87-4305b3b340aa.png)

**Reply by ukutaht** [reply]
Fascinating! I'm definitely interested in getting to the bottom of this. Let me know which one it is if you figure it out!

**Comment by MagnusThome** [comment]
A very strange observation that might be a lead to something:

I've run the plausible.js all the time to the cloud hosted Plausible. And yesterday I changed from plausible.outbound-links.js to plausible.js to my local hosted Plausible.

But on my cloud hosted web page with logging data I have data for outbound clicks in yesterday's data! This popped up on the cloud page after I removed outbound click logging for my **local** hosted. I did not touch the cloud javascript links in the web pages what so ever! 

Scrolling through data for different days there's  **only** click data available for yesterday on the cloud. And how many clicks it shows suggests it's clicks for part of the day yesterday. Sort of matching from after I removed the outbound links javascript for my local host.  Veeeeery odd behaviour :-)

**Comment by MagnusThome** [comment]
After not using outbound links js at all I added it back to my self host link and not to the cloud host link. 

After this change I had outbound click statistics for the rest of the day on the wrong host! My local host had none. And the cloud server showed outbound clicks. 


Oooops :-)

**Comment by MagnusThome** [comment]
BTW, love what you have put together, it's really really nice and also just exactly what I want and need 👍

**Comment by MagnusThome** [comment]
I understand the script isn't written for two instances being on the page at the same time, one for self hosting and one for cloud. There is no case for that obviously (apart from this "odd" test I did).

So the question is, is the noticeable discrepancy in pageviews (in the opposite direction than expected) due to a clash between the dual scripts or something else? I don't know of course. But my findings are these:

Going from port 8000 to normal https does make a difference in pageviews. I flipped between these two and it looked very clear that there was a difference. Using port 8000 the local logs were visibly less, with it removed local logs were instead just slightly more than the cloud values. 

I didn't test much on using the outbound javascript or not due to the odd behavior showing click data on the wrong host. 

I also tested placing the self host script line in the web code before the cloud hosted script line and the other way around. With limited testing I still have one observation on this:

using port 8000 + outbound script + cloud host script line placed first = gave the largest difference (local data being lower) 

using port 8000 + normal script + self host script placed first = gave a visible but not as large difference (local data being lower) 

using normal https with subdomain + normal script + self host script placed first = data looked very good, local data just a tiny tiny bit higher than the cloud data. 

As you can see some test scenarios are missing. Didn't think it was worth spending time on these due to the odd behaviour of click data being found on the wrong host of the two.

**Reply by ukutaht** [reply]
Thanks for testing out the different scenarios. I'm still quite confused by this though, does feel like some sort of conflict between the two scripts being on the same page.

**Reply by MagnusThome** [reply]
Yeah, agree. Not sure 100% all the discrepancies are from that though. But surely there are some odditities there :-)

**Reply by Daniel15** [reply]
Some very restrictive firewalls (for example, at schools and workplaces) can block ports other than 80 and 443, which might be the cause of some of the discrepancies. 

(sorry, I know this discussion is over a year old now)

**Comment by MagnusThome** [comment]
Announcement throwing out Google Analytics 

https://rejsa.nu/forum/viewtopic.php?t=122957

**Reply by metmarkosaric** [reply]
nice!
